src/lib/world/World.ts

```diff
@@ -96,6 +98,14 @@ class World {
         this.desktopManager.getCurrentDesktop().arrange();
     }
 
+    public doGesture(progress: number) {
+        this.desktopManager.getCurrentDesktop().gestureScroll(progress);
+    }
+
+    public finishGesture() {
+        this.desktopManager.getCurrentDesktop().finishGesture();
+    }
```

Comment:
```suggestion
    public gestureScroll(amount: number) {
        this.do((clientManager, desktopManager) => desktopManager.getCurrentDesktop().gestureScroll(amount));
    }

    public gestureScrollFinish() {
        this.do((clientManager, desktopManager) => desktopManager.getCurrentDesktop().gestureScrollFinish());
    }
```

Avoid using `desktopManager` directly on user actions. Use `World.do` instead. This automatically calls `arrange` afterwards, so you can remove it in `Desktop.gestureScroll`.

I've also added some renaming suggestions to allow more gestures in the future.